src/app/wallets/update-pending-payments.ts

```diff
@@ -0,0 +1,187 @@
+    return lnPaymentLookup
+  }
+  const { status, roundedUpFee } = lnPaymentLookup
+
+  if (status === PaymentStatus.Settled || status === PaymentStatus.Failed) {
+    const ledgerService = LedgerService()
+    const settled = await ledgerService.settlePendingLiabilityTransactions(paymentHash)
+    if (settled instanceof Error) {
+      paymentLogger.error({ error: settled }, "we didn't have any transaction to update")
+      return settled
+    }
+
+    // PaymentStatus.Failed transactions have null 'roundedUpFee'
+    if (status === PaymentStatus.Settled && roundedUpFee !== null) {
+      paymentLogger.info(
+        { success: true, id: paymentHash, payment: paymentLiabilityTx },
+        "payment has been confirmed",
+      )
+      return reimburseFee({
+        liabilitiesAccountId,
+        paymentLiabilityTx,
+        roundedUpFee,
+        logger,
+      })
+    }
+    return revertTransaction({
+      paymentLiabilityTx,
+      lnPaymentLookup,
+      logger: paymentLogger,
+    })
+  }
+}
+
+const reimburseFee = async ({
+  liabilitiesAccountId,
+  paymentLiabilityTx,
+  roundedUpFee,
+  logger,
+}: {
+  liabilitiesAccountId: LiabilitiesAccountId
+  paymentLiabilityTx: LedgerTransaction
+  roundedUpFee: Satoshis
+  logger: Logger
+}): Promise<void | ApplicationError> => {
+  const { paymentHash } = paymentLiabilityTx
+  // If we had PaymentLedgerType => no need for checking the fields
+  if (!paymentHash)
+    return new InconsistentDataError("paymentHash missing from payment transaction")
+
+  if (!paymentLiabilityTx.feeKnownInAdvance) {
+    const feeDifference = FeeReimbursement(paymentLiabilityTx.fee).getReimbursement({
+      actualFee: roundedUpFee,
+    })
+    if (feeDifference === null) {
+      logger.warn(
+        `Invalid reimbursement fee for ${{
+          maxFee: paymentLiabilityTx.fee,
+          actualFee: roundedUpFee,
+        }}`,
+      )
+      return
+    }
+
+    logger.info(
+      {
+        paymentResult: paymentLiabilityTx,
+        feeDifference,
+        maxFee: paymentLiabilityTx.fee,
+        actualFee: roundedUpFee,
+        id: paymentHash,
+      },
+      "logging a fee difference",
+    )
+
+    const price = await PriceService().getCurrentPrice()
+    if (price instanceof Error) return price
+    const usd = feeDifference * price
+
+    const ledgerService = LedgerService()
+    const result = await ledgerService.receiveLnFeeReimbursement({
+      liabilitiesAccountId,
+      paymentHash: paymentHash,
+      sats: feeDifference,
+      usd,
+      journalId: paymentLiabilityTx.journalId,
+    })
+    if (result instanceof Error) return result
+  }
+}
+
+const revertTransaction = async ({
+  paymentLiabilityTx,
+  lnPaymentLookup,
+  logger,
+}: {
+  paymentLiabilityTx: LedgerTransaction
+  lnPaymentLookup: LnPaymentLookup
+  logger: Logger
+}): Promise<void | ApplicationError> => {
+  const ledgerService = LedgerService()
+  const voided = await ledgerService.voidLedgerTransactionsForJournal(
```

Comment:
Journal is a concept of Medici, do we really want to keep that concept?

Reply:
If we can keep it internal to the `ledgerService` that would be great - but how do we then identify the transactions? `PaymentHash`?

Reply:
yep, the solution is use payment hash to query one transaction, get the journalId and do the void. (we can leave this for db migration)